src/main/java/com/hivemq/cli/commands/cli/TestBrokerCommand.java

```diff
@@ -0,0 +1,368 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.hivemq.cli.commands.cli;
+
+import com.google.common.base.Joiner;
+import com.hivemq.cli.DefaultCLIProperties;
+import com.hivemq.cli.commands.options.AuthenticationOptions;
+import com.hivemq.cli.commands.options.SslOptions;
+import com.hivemq.cli.converters.MqttVersionConverter;
+import com.hivemq.cli.mqtt.test.Mqtt3FeatureTester;
+import com.hivemq.cli.mqtt.test.Mqtt5FeatureTester;
+import com.hivemq.cli.mqtt.test.results.*;
+import com.hivemq.client.mqtt.MqttClientSslConfig;
+import com.hivemq.client.mqtt.MqttVersion;
+import com.hivemq.client.mqtt.datatypes.MqttQos;
+import com.hivemq.client.mqtt.mqtt3.message.connect.connack.Mqtt3ConnAck;
+import com.hivemq.client.mqtt.mqtt3.message.connect.connack.Mqtt3ConnAckReturnCode;
+import com.hivemq.client.mqtt.mqtt5.message.connect.connack.Mqtt5ConnAck;
+import com.hivemq.client.mqtt.mqtt5.message.connect.connack.Mqtt5ConnAckReasonCode;
+import com.hivemq.client.mqtt.mqtt5.message.connect.connack.Mqtt5ConnAckRestrictions;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.tinylog.Logger;
+import org.tinylog.configuration.Configuration;
+import picocli.CommandLine;
+
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@CommandLine.Command(
+        name = "test",
+        description = "Tests the specified broker on different MQTT feature support and prints the results")
+public class TestBrokerCommand implements Runnable {
+
+    final int MAX_PAYLOAD_TEST_SIZE = 100000; // ~ 1 MB
+
+    @CommandLine.Option(names = {"-h", "--host"}, description = "The hostname of the message broker (default 'localhost')", order = 1)
+    private @Nullable String host;
+
+    @CommandLine.Option(names = {"--help"}, usageHelp = true, description = "display this help message")
+    boolean usageHelpRequested;
+
+    @CommandLine.Option(names = {"-p", "--port"}, description = "The port of the message broker (default: 1883)", order = 1)
+    private @Nullable Integer port;
+
+    @CommandLine.Option(names = {"-V", "--mqttVersion"}, converter = MqttVersionConverter.class, description = "The mqtt version to test the broker on", order = 1)
+    private @Nullable MqttVersion version;
+
+    @CommandLine.Option(names = {"-f", "--force"}, defaultValue = "false", description = "Force test MQTT 5 features (even if MQTT3 features were already tested)", order = 1)
```

Comment:
I think force is the right word here, maybe use --full, and as description: Perform all tests for all MQTT versions

Reply:
Fully agree. What do you think about `-a` and '--all'?

Reply:
fine for me